Source/WebCore/rendering/svg/SVGTextLayoutAttributesBuilder.cpp

```diff
@@ -204,31 +196,17 @@ void SVGTextLayoutAttributesBuilder::fillCharacterDataMap(const TextPosition& po
         if (!xListPtr && !yListPtr && !dxListPtr && !dyListPtr && !rotateListPtr)
             break;
 
-        SVGCharacterDataMap::iterator it = m_characterDataMap.find(position.start + i + 1);
-        if (it == m_characterDataMap.end()) {
-            SVGCharacterData data;
-            updateCharacterData(i, lastRotation, data, lengthContext, xListPtr, yListPtr, dxListPtr, dyListPtr, rotateListPtr);
-            m_characterDataMap.set(position.start + i + 1, data);
-            continue;
-        }
-
-        updateCharacterData(i, lastRotation, it->value, lengthContext, xListPtr, yListPtr, dxListPtr, dyListPtr, rotateListPtr);
+        auto& data = m_characterDataMap.add((position.start + i + 1), SVGCharacterData()).iterator->value;
```

Comment:
We should use `HashMap::ensure()` to avoid creating a useless `SVGCharacterData()` object when the key is already there.